### DISCUSSION TITLE
Single disk - Zero Parity mode - bitrot protection available?

### DISCUSSION THREAD

**Discussion by jstiops** [minio/minio#15593] [Q&A] [answered]
i asked arround in slack, did not get reply. Hope to get one here. 
I really want to know if bitrot protection (highway hashing) is enabled in single disk single node mode , using the new zero parity things that came with a recent minio verion. The Zero Parity feature enabled some features even though no EC parity was done on a single disk. So i was wondering if bitrot protection would still run on writing/reading objects in that mode or if it's only enabled when EC is running with proper parity for example.

**Comment by eco-minio** [comment]
As mentioned in Slack, it is not provided via a single disk. It should be thought of only as feature enablement for things like replication. In a single disk setup you can only derive protection for an external system such as a NAS/RAID. In all cases, durability guarantees on a single drive are misleading as all of those guarantees go out the window if the drive dies.

**Reply by jstiops** [reply]
sorry, but i'm still trying to understand the concepts properly. 
I see everything related EC (like parity, replication etc.) as a data **redundancy** or **protection** concept. I see the bitrot protection feature as a data **integrity** concept. In case of single disk, i can see the former part being handled by underlying RAID, but that does not provide the same integrity protection as the latter. That's why i would wonder why bitrot protection with the highway hashing would not be available in single disk, because it's a different process compared to EC-related things. 

They have different goals. failing drives are handled by EC (so obviously not available/active with single disk), but bitrot protection has nothing to do with that hw failure scenario and is there to ensure the data is correct in case hardware is fine.

Is my thinking wrong?

**Comment by klauspost** [comment, marked as answer]
Yes, bitrot checks are performed in Single-Node Single-Drive mode.

Of course on failures cannot be recovered.

**Reply by jstiops** [reply]
understood. thanks!
